website/docs/cli/commands/show.html.md

```diff
@@ -28,6 +28,11 @@ For Terraform state files (including when no path is provided),
 For Terraform plan files, `terraform show -json` will show a JSON representation
 of the plan, configuration, and current state.
 
+In order to properly json encode the state, the state must match the schemas
+for the installed providers. To display the current state, the provider schemas
+must not have changed since the previous `terraform apply`. To display a
```

Comment:
🤔 This feels a bit too passive, and it doesn't cover what to do if you're just showing state but have updated providers. Here's a pseudo-documentation example:

If you've updated provider versions since the state was written, the state needs to be upgraded before it can be displayed with `show -json`. If you are viewing a plan, it must be created without `-refresh=false`. If you are viewing a statefile, run `terraform refresh` first.

I am not suggesting that ^^ exactly, but perhaps it'll give you some ideas?

Reply:
I agree making this more active reads better. I'll update that commit